### PR TITLE
Usability tweaks

### PR DESCRIPTION
**Pull request by ximon18**
Various trait impls and other tweaks to enable more use cases.

Based on code that was originally in the `xfr` branch for now closed PR #335 which used these changes to implement the `ZoneMaintainer`.

### CODE REVIEW

File: src/tsig/mod.rs [hunk 1]
```diff
@@ -389,6 +393,15 @@ impl AsRef<Key> for Key {
     }
 }
 
+//--- Display
+
+#[cfg(feature = "std")]
+impl fmt::Display for Key {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_fmt(format_args!("{}", self.name))
```
Comment (partim) [inline src/tsig/mod.rs:401]: This is equivalent to

```
write!(f, "{}", self.name)
```

or probably even just `self.name.fmt(f)`.

File: src/zonetree/update.rs [hunk 2]
```diff
@@ -1309,6 +1309,21 @@ pub enum Error {
...
 
+//--- Display
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Error::OutOfZone => f.write_str("OutOfZone"),
+            Error::NotSoaRecord => f.write_str("NotSoaRecord"),
+            Error::IoError(err) => {
+                f.write_fmt(format_args!("I/O error: {err}"))
```
Comment (partim) [inline src/zonetree/update.rs:1320]: As above: Probably clearer to use `write!()`.

File: src/zonetree/zone.rs [hunk 3]
```diff
@@ -13,9 +13,47 @@ use super::traits::WritableZone;
...
 use super::{parsed, ReadableZone, ZoneStore};
 
+//------------ ZoneKey -------------------------------------------------------
+
+/// A key that uniquely identifies a zone.
+///
+/// A zone is identified by the owner name of the apex and its class. Every
+/// record in a zone must be at or under the apex owner name and be of the
+/// same class.
+pub type ZoneKey = (StoredName, Class);
```
Comment (partim) [inline src/zonetree/zone.rs:23]: `ZoneKey` is potentially confusing what with DNSSEC also using zone keys …

Comment (ximon18) [reply src/zonetree/zone.rs:23]: Good point.

Comment (ximon18) [reply src/zonetree/zone.rs]: Renamed to `ZoneTuple`.

**Review by partim — APPROVED** [review]

**Pull request merged**